# Generated by Django 4.2.3 on 2023-09-22 09:37

from django.conf import settings
from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    dependencies = [
        ("accounts", "0051_merge_20230912_0748"),
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
        ("google", "0001_initial"),
    ]

    operations = [
        migrations.CreateModel(
            name="GoogleSyncErrors",
            fields=[
                (
                    "id",
                    models.BigAutoField(
                        auto_created=True,
                        primary_key=True,
                        serialize=False,
                        verbose_name="ID",
                    ),
                ),
                (
                    "on_task",
                    models.CharField(
                        choices=[
                            ("create_user", "Create user"),
                            ("update_user", "Update user"),
                            ("suspend_user", "Suspend user"),
                            ("create_group", "Create group"),
                            ("update_group", "Update group"),
                        ],
                        max_length=50,
                    ),
                ),
                ("task_kwargs", models.JSONField(blank=True, null=True)),
                ("error", models.TextField()),
                ("created_at", models.DateTimeField(auto_now_add=True)),
                (
                    "people_group",
                    models.ForeignKey(
                        blank=True,
                        null=True,
                        on_delete=django.db.models.deletion.CASCADE,
                        related_name="google_sync_errors",
                        to="accounts.peoplegroup",
                    ),
                ),
                (
                    "user",
                    models.ForeignKey(
                        blank=True,
                        null=True,
                        on_delete=django.db.models.deletion.CASCADE,
                        related_name="google_sync_errors",
                        to=settings.AUTH_USER_MODEL,
                    ),
                ),
            ],
            options={
                "verbose_name": "google sync error",
                "verbose_name_plural": "google sync errors",
                "ordering": ("-created_at",),
            },
        ),
    ]
